specification/dartLangSpec.tex

```diff
@@ -13969,8 +15441,12 @@ \subsection{Lexical Lookup}
   (\ref{classes}).
 \item
   When $\ell$ has access to \THIS{}
-  it is a compile-time error if the interface of the enclosing class
-  does not have a member named $n$.
+  it is a compile-time error if $\ell$ occurs in
+  a class \DefineSymbol{C} and
+  the interface of $C$ does not have a member named $n$;
+  and it is a compile-time error if $\ell$ occurs in
+  an extension \DefineSymbol{E},
+  and the interface of the \ON{} type of $E$ does not have a member named $n$.
```

Comment:
@eernstg  I don't think this is correct, unless I'm missing something.  Consider:

```dart
extension F on int {
  void foo() {}
}

extension G on int {
  void bar() {
    foo();
  }
}
```

I believe this implies that the call to `foo()` inside of `bar` should be an error, right?

Reply:
Good catch! We should indeed take implicit extension member invocations into account here.

Cf. https://github.com/dart-lang/language/pull/1501.

I'll close this thread because the discussion can be handled in the new PR.